src/platform_impl/windows/event_loop.rs

```diff
@@ -2072,23 +2077,7 @@ unsafe fn public_window_callback_inner<T: 'static>(
     }
 
     win32wm::WM_SETTINGCHANGE => {
-      use crate::event::WindowEvent::ThemeChanged;
-
-      let preferred_theme = subclass_input.window_state.lock().preferred_theme;
-      let new_theme = try_window_theme(
-        window,
-        preferred_theme.or(*subclass_input.event_loop_preferred_theme.lock()),
-      );
-      let mut window_state = subclass_input.window_state.lock();
-
-      if window_state.current_theme != new_theme {
-        window_state.current_theme = new_theme;
-        mem::drop(window_state);
-        subclass_input.send_event(Event::WindowEvent {
-          window_id: RootWindowId(WindowId(window.0 as _)),
-          event: ThemeChanged(new_theme),
-        });
-      }
```

Comment:
I think we should keep this one separate and add back the check that discarded unnecessary logic which was

```rs
if preferred_theme.or(*subclass_input.event_loop_preferred_theme.lock()).is_none() {
  // try set new theme based on OS
  // ...etc
}
```

Reply:
or keep the new function but pass it a flag to know that it is called from `WM_SETTINGCHANGE` and skip above check matches
```rs
if from_os && preferred_theme.or(*subclass_input.event_loop_preferred_theme.lock()).is_none() {
  return;
}
```